app/Models/Emulator.php

```diff
@@ -148,6 +149,33 @@ public function registerMediaCollections(): void
 
     // == accessors
 
+    public function getHasOfficialSupportAttribute(): bool
+    {
+        // Officially supported emulators have at least one user agent registered in the DB.
+        if ($this->userAgents->isEmpty()) {
+            return false;
+        }
+
+        foreach ($this->userAgents as $userAgent) {
+            // Case 1: Has minimum_hardcore_version. This implies there's at least one
+            // version of the emulator out there which we fully support.
+            if ($userAgent->minimum_hardcore_version) {
+                return true;
+            }
+
+            // Case 2: Emulator is active and either doesn't have minimum version requirements
+            // or meets the minimum allowed version requirement.
+            if (
+                $this->active
+                && (!$userAgent->minimum_allowed_version || $userAgent->minimum_allowed_version)
```

Comment:
Am I reading this correctly? && `(!A || A)`? Won't that always be true?

Reply:
Good catch - I made a typo here. With a fresh set of eyes, I've changed up this logic in latest.

I used this code for local debugging:
```php
$emulators = Emulator::with("userAgents")->get();

$results = [];

foreach ($emulators as $emulator) {
  $results[] = [
    "id" => $emulator->id,
    "name" => $emulator->name,
    "has_official_support" => $emulator->has_official_support
  ];
}

dd($results);
```